vsphere problem detector (OSDOCS-1731) (#28260)
openshift/openshift-docs: modules/understanding-vsphere-problem-detector.adoc

// Module included in the following assemblies:
//
// * installing/installing_vsphere/using-vsphere-problem-detector-operator.adoc

> :operator-name: vSphere Problem Detector Operator

Author: I could swear I heard or read not to use a substitution in a heading.  I used an attribute because A) I wasn't sure of the final name when I started, and B) I grew tired of the idea of rekeying it each time.

> :operator-name: vSphere Problem Detector Operator

Reviewer: Fair enough. And no an attribute in a heading shouldn't be a problem. Just searched the repo (`grep -r "^=.*{" | wc -l`) and see 280 hits across existing modules that use 'em.

> :operator-name: vSphere Problem Detector Operator

Author: Then I'll pop it in the heading.  I prefer consistency but thought I was cornered into _not_ using the attr.

[id="understanding-vsphere-problem-detector_{context}"]

> = Understanding the vSphere Problem Detector Operator

Reviewer 2: I suspect Mike was kindly adhering to the previously set conventions used by a couple of other Storage assemblies. Wouldn't be a bad idea to update those and this one to "About..." at some point.

The {operator-name} checks clusters that are deployed on vSphere for common installation and misconfiguration issues that are related to storage.

> The Operator runs in the `openshift-cluster-storage-operator` namespace and is started by the `cluster-storage-operator` when the Cluster Storage Operator detects that the cluster is deployed on vSphere. The {operator-name} communicates with the vSphere vCenter Server to determine the virtual machines in the cluster, the default datastore, and so on. The Operator uses the credentials from the `cloud-credentials-provider` to connect to vSphere.

Author: Thank you.

> The Operator runs in the `openshift-cluster-storage-operator` namespace and is started by the `cluster-storage-operator` when the Cluster Storage Operator detects that the cluster is deployed on vSphere. The {operator-name} communicates with the vSphere vCenter Server to determine the virtual machines in the cluster, the default datastore, and other information about the vSphere vCenter Server configuration. The Operator uses the credentials from the `cloud-credentials-operator` to connect to vSphere.

Author: How did I miss that?

> 

Reviewer 3: credentials from the `cloud credentials operator`?

Author: @jsafrane  is Ping correct or is it "vsphere-cloud-credentials"?  My interest is to let customers know which credentials to check if the Operator cannot communicate with vSphere.

Author: @gnufied , Bob told me that Jan might not be around for a bit.  Is this something you can clarify?

Can you share how to get the correct answer so that I can be more effective in the future?  Thank you.

Reviewer 4: It's cloud-credential-operator, https://github.com/openshift/cloud-credential-operator

> The Operator runs the checks periodically. The schedule is as follows:

Reviewer 5: ```suggestion
The Operator runs the checks according to the following schedule:
```

Author: Much better.

* The checks run every 8 hours.

* If any check fails, the Operator runs the checks again in intervals of 1 minute, 2 minutes, 4, 8, and so on. The Operator doubles the interval up to a maximum interval of 8 hours.

* When all checks pass, the schedule returns to an 8 hour interval.

The Operator increases the frequency of the checks after a failure so that the Operator can report success quickly after the failure condition is remedied. You can run the Operator manually for immediate troubleshooting information.

// Clear temporary attributes

> :operator-name:

Reviewer 5: ```suggestion
:!operator-name:
```
You need an exclamation point in the statement to unset an attribute. See http://asciidoc.github.io/asciidoc/chunked/ch28.html

Author: Many thanks.  I hallucinated some doc that suggested setting it to empty was equivalent.
